### ISSUE TITLE
Leaderboard rankings jumbled up

### ISSUE THREAD

**Issue by spMohanty** [crowdAI/crowdai#932]
<img width="732" alt="screen shot 2018-08-14 at 16 46 42" src="https://user-images.githubusercontent.com/1581312/44098904-b17a921c-9fe1-11e8-91e9-d7973d695880.png">

This happens when I click on "load more".

**Comment by scarroll32** [comment]
After much investigating, I've determined this issue was caused by the recent "leaderboard baseline" feature, and is hopefully restricted to the Vizdoom challenge anyway.

In the Vizdoom challenge there are multiple people with the same scores. We sort on **score** and **score_secondary**. 

<img width="483" alt="screen shot 2018-08-16 at 14 38 19" src="https://user-images.githubusercontent.com/11340230/44209051-8381f800-a162-11e8-9252-b6e80b54543f.png">

With the leaderboard baseline we introduced a second sort on the same data, and it works like this:

1) Rank the participants by score, excluding the baseline rows, to assign rankings and change of ranking
2) Insert the baseline rows
3) Sort all rows again

Even though the Vizdoom challenge had no baseline, this problem occurs as we are sorting twice on data that is not different, and without provide a distinguishing sort attribute, so it is non-deterministic.

**Solution**

Add an overriding sort field on the calculated row_num to use as a consistent tiebreaker.
